langlib/langlib-test/src/test/resources/test-src/intlib_test.bal

```diff
@@ -123,3 +123,113 @@ function testIntOverflow() {
     test:assertValueEqual("{ballerina/lang.int}NumberOverflow", err.message());
     test:assertValueEqual("int range overflow", <string>checkpanic err.detail()["message"]);
 }
+
+function testIntOverflowWithSum() {
+    int a1 = int:MIN_VALUE;
+    int|error result = trap a1.sum(-1);
```

Comment:
Since, `a1.sum()` and `int:sum()` calls to the same internal function, checking for both cases for this fix can be redundant. Shall we have only one of these?

Reply:
But it's two ways calling the same function right? I believe it's okay to have some tests with both.

Reply:
Yeah. It's okay to have them both. +1
My concern is this fix is specific to the method implementation and the way of calling it does not get impacted.

Reply:
We don't really have many tests covering the behavior of lang.int functions. That's why I decided to add tests for both. :)

Reply:
IMO when adding tests for langlib methods it is better to add tests for both ways of calling them. Sometimes we may come across issues with how the first argument is propagated/analyzed.